.github/workflows/integrate-rock.yaml

```
@@ -0,0 +1,113 @@
+name: Integrate ROCK into charm
+
+on:
+  workflow_call:
+    inputs:
+      rock-dir:
+        required: true
+        type: string
+      event-name:
+        required: true
+        type: string
+      full-image-tag:
+        description: "Full image reference to the built and pushed rock image"
+        required: true
+        type: string
+
+  workflow_dispatch:
+    inputs:
+      rock-dir:
+        description: "Path to the rock directory (e.g. rocks/my-rock)"
+        required: true
+        type: string
+      target-branch:
+        description: "Branch to integrate into"
+        required: true
+        type: string
+      dry-run:
+        description: "Run in dry-run mode?"
+        required: false
+        default: true
+        type: boolean
+      full-image-tag:
+        description: "Full image reference from Docker registry"
+        required: true
+        type: string
+
+jobs:
+  integrate-rock:
+    name: Integrate ${{ inputs.full-image-tag }}
+    runs-on: ubuntu-24.04
+    env:
+      GH_TOKEN: ${{ secrets.GH_TOKEN }}
+      GH_USER_EMAIL: ${{ secrets.GH_USER_EMAIL }}
+
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v4
+
+      - name: Install charmed-analytics-ci
+        run: pip install charmed-analytics-ci==1.1.0
```

Since we're using this as an executable, it is preferable to use `pipx` for this action
```suggestion
        run: pipx install charmed-analytics-ci==1.1.0
```